### ISSUE TITLE
Not a number

### ISSUE THREAD

**Issue by vladimirnewissue** [paradoxxxzero/gnome-shell-system-monitor-applet#594]
Disk section shows "not a number" and filled with color. Screenshot below.

![notanumber](https://user-images.githubusercontent.com/64399031/80342895-a9372e00-886d-11ea-8413-43126162a83e.png)

**Comment by farhadinima75** [comment]
same problem after today Update.... :(

**Comment by AlexFalappa** [comment]
Can confirm this after update on 27 Apr 2020.

**Comment by vladimirnewissue** [comment]
> Can confirm this after update on 27 Apr 2020.
> 
> I am on Ubuntu 18.04.
> 
> Any workaround?

me too ubuntu 18.04. no workaround yet...

**Comment by ivocavalcante** [comment]
Same here, Ubuntu 18.04. Disk rates and Temp show NaN. Also, this is what temperature source selector shows:

![image](https://user-images.githubusercontent.com/11679725/80493479-d94d0280-893b-11ea-873d-ef6202581525.png)

**Comment by techsy730** [comment]
For the Disk issue, I basically undid https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/commit/3ba2f60c45eac1fce8702387884f96840e224db2

Replace
`let lines = ByteArray.toString(as_r[1]).split('\n');`
with
`let lines = String(as_r[1]).split('\n');`

For the Temperature issue, the exact same fix described in
https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/issues/520#issuecomment-498990185 works here too.

**Comment by vladimirnewissue** [comment]
> For the Disk issue, I basically undid [3ba2f60](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/commit/3ba2f60c45eac1fce8702387884f96840e224db2)
> 
> In `~/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js`
> Find the section
> 
> ```js
>         let file = Gio.file_new_for_path('/proc/diskstats');
>         file.load_contents_async(null, (source, result) => {
>             let as_r = source.load_contents_finish(result);
>             let lines = ByteArray.toString(as_r[1]).split('\n');
> ```
> 
> Replace the last line:
> 
> ```js
> let lines = ByteArray.toString(as_r[1]).split('\n');
> ```
> 
> with
> 
> ```js
> let lines = String(as_r[1]).split('\n');
> ```
> 
> For the Temperature issue, the exact same fix described in
> [#520 (comment)](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/issues/520#issuecomment-498990185) works here too.

Great! Thanks!

**Comment by chrisspen** [comment]
@techsy730 I just tested this fix, and unfortunately it does not resolve the problem for Gnome Shell 3.28.

**Comment by vladimirnewissue** [comment]
> @techsy730 I just tested this fix, and unfortunately it does not resolve the problem for Gnome Shell 3.28.

mine is 3.28.4 and the problem is solved

**Comment by techsy730** [comment]
> @techsy730 I just tested this fix, and unfortunately it does not resolve the problem for Gnome Shell 3.28.

Did you log out and log back in? (Merely disabling and re-enabling doesn't pick up the new code).
Forgot to mention that in my instructions.

**Comment by ncallister** [comment]
> > @techsy730 I just tested this fix, and unfortunately it does not resolve the problem for Gnome Shell 3.28.
> 
> Did you log out and log back in? (Merely disabling and re-enabling doesn't pick up the new code).
> Forgot to mention that in my instructions.

If you don't want to log out, reloading gnome (ALT+F2, r) works too.

**Comment by cmcinroy** [comment]
Fixed for me, Ubuntu 18.04.4 GNOME Shell 3.28.4.
Thanks!

**Comment by acesabe** [comment]
Fixes for Disk R/W but not for Temp and Fan....

**Comment by techsy730** [comment]
> Fixes for Disk R/W but not for Temp and Fan....

Yes, for Temp and Fan you will need to follow the instructions at
https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/issues/520#issuecomment-498990185

**Comment by hdavid0510** [comment]
> For the disk issue, I basically undid [3ba2f60](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/commit/3ba2f60c45eac1fce8702387884f96840e224db2)
> 
> In `~/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js`
> Find the section
> 
> ```js
>         let file = Gio.file_new_for_path('/proc/diskstats');
>         file.load_contents_async(null, (source, result) => {
>             let as_r = source.load_contents_finish(result);
>             let lines = ByteArray.toString(as_r[1]).split('\n');
> ```
> 
> Replace the last line:
> 
> ```js
> let lines = ByteArray.toString(as_r[1]).split('\n');
> ```
> 
> with
> 
> ```js
> let lines = String(as_r[1]).split('\n');
> ```
> 
> Then restart GNOME, either with ALT+F2, r, or logging out and logging back in (only unloading and reloading the extension doesn't work)
> 
> This was done on Gnome Shell 3.28.4
> 
> For the Temperature issue, the exact same fix described in
> [#520 (comment)](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/issues/520#issuecomment-498990185) works here too.

Great, this worked like a charm.
Just like disk one, temperature shown as NaN also fixed with this way - replacing `ByteArray.toString` to `String`
Thanks a lot!

**Comment by frankvw2017** [comment]
Same problem here. Disk activity shown as NaN. Running GNOME Shell 3.28.4

**Comment by peci1** [comment]
The last mentioned fix works for me on GNOME Shell 3.28.4. Locking screen works without side-effects.
